opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -168,17 +253,18 @@ public function ajaxValidateCredentials() {
 
         // If validated, save validated status to db.
         if (array_key_exists('success', $validate)) {
-            $this->load->model('setting/setting');
+            // Load Smaily admin model.
+            $this->load->model('smailyforopencart/admin');
             // Used because save button saves whole form.
             $settings = [
-                'smaily_api_credentials' => [
-                    'password' => $password,
-                    'subdomain' => $subdomain,
-                    'username' => $username,
-                ],
+            'smaily_api_credentials' => [
```

Comment:
Invalid indentation.

But you can refactor this to:
```php
$settings = [
  'password' => $password,
  'subdomain' => $subdomain,
  'username' => $username,
];
```

Then you do not need to use `$settings['smaily_api_credentials']` to access the data when updating database.